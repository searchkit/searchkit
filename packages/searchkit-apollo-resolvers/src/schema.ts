import gql from 'graphql-tag'

export default gql`
  type Hit {
    id: String
    fields: HitFields
  }

  type SortOption {
    id: String
    label: String
  }

  type Summary {
    total: Float
    appliedFilters: [SelectedFilter]
    query: String
    sortOptions: [SortOption]
  }

  type SelectedFilter {
    id: String
    label: String
    value: String
  }

  type ResultSet {
    summary: Summary
    hits(page: PageInput, sortBy: String): HitResults
    facets: [FacetSet]
    facet(id: String!, query: String, page: PageInput): FacetSet
  }

  type PageInfo {
    total: Float
    totalPages: Float
    pageNumber: Float
    from: Float
    size: Float
  }

  type HitResults {
    items: [Hit]
    page: PageInfo
    sortedBy: String
  }

  input PageInput {
    from: Float
    size: Float
  }

  input FiltersSet {
    id: String
    value: String
    min: Float
    max: Float
    dateMin: String
    dateMax: String
  }

  interface FacetSet {
    id: String
    label: String
    type: String
    display: String
    entries: [FacetSetEntry]
  }

  type FacetSetEntry {
    id: String
    label: String
    count: Float
    isSelected: Boolean
  }

  type RefinementSelectFacet implements FacetSet {
    id: String
    label: String
    type: String
    display: String
    entries: [FacetSetEntry]
  }

  type RangeFacet implements FacetSet {
    id: String
    label: String
    type: String
    display: String
    entries: [FacetSetEntry]
  }

  type DateRangeFacet implements FacetSet {
    id: String
    label: String
    type: String
    display: String
    entries: [FacetSetEntry]
  }

  extend type Query {
    results(query: String, filters: [FiltersSet], page: PageInput): ResultSet
  }
`
