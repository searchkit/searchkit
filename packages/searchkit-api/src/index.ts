import type { MultipleQueriesQuery as AlgoliaMultipleQueriesQuery } from '@algolia/client-search'
import { transformRequest } from './transformRequest'
import transformResponse, { transformFacetValuesResponse } from './transformResponse'
import { ClientConfig, SearchRequest, RequestOptions, Transporter, AppSettings } from './types'
import { ESTransporter } from './Transporter'
import { getQueryRulesActionsFromRequest, QueryRuleActions } from './queryRules'
import { createElasticsearchQueryFromRequest } from './utils'
export * from './types'

class Client {
  transporter: Transporter

  constructor(private config: ClientConfig, private settings: AppSettings = { debug: false }) {
    this.transporter =
      'msearch' in config.connection ? config.connection : new ESTransporter(config.connection)
  }

  private async performSearch(requests: SearchRequest[]) {
    if (this.settings.debug) {
      console.log('Performing search with requests:')
      console.log('POST /_msearch')
      console.log(createElasticsearchQueryFromRequest(requests))
    }
    const responses = await this.transporter.msearch(requests)
    return responses
  }

  async handleRequest(body: any, requestOptions?: RequestOptions) {
    const instantsearchRequests = body as AlgoliaMultipleQueriesQuery[]

    const instantsearchResponses = this.handleInstantSearchRequests(
      instantsearchRequests,
      requestOptions
    )

    return instantsearchResponses
  }

  async handleInstantSearchRequests(
    instantsearchRequests: AlgoliaMultipleQueriesQuery[],
    requestOptions?: RequestOptions
  ) {
    const queryRules = this.config.search_settings.query_rules || []

    const requestQueryRuleActions: QueryRuleActions[] = instantsearchRequests.map((request) => {
      return getQueryRulesActionsFromRequest(queryRules, request)
    })

    let esRequests: SearchRequest[] = instantsearchRequests.map((request, i) => ({
      body: transformRequest(
        request,
        this.config.search_settings,
        requestQueryRuleActions[i],
        requestOptions
      ),
      request: request,
      indexName: request.indexName
    }))

    if (requestOptions?.hooks?.beforeSearch) {
      esRequests = await requestOptions.hooks.beforeSearch(esRequests)
    }

    let esResponses = await this.performSearch(esRequests)

    if (requestOptions?.hooks?.afterSearch) {
      esResponses = await requestOptions.hooks.afterSearch(esRequests, esResponses)
    }

    const instantsearchResponses = esResponses.map((response, i) => {
      // @ts-ignore
      if (instantsearchRequests[i].params?.facetName) {
        return transformFacetValuesResponse(response, instantsearchRequests[i])
      }
      return transformResponse(
        response,
        instantsearchRequests[i],
        this.config.search_settings,
        requestQueryRuleActions[i]
      )
    })

    return {
      results: instantsearchResponses
    }
  }
}

const createClient = (config: ClientConfig, settings?: AppSettings) => new Client(config, settings)

export default createClient
