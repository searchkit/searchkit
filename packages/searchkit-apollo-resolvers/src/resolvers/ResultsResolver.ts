import QueryManager, { MixedFilter } from '../core/QueryManager'
import SearchkitRequest from '../core/SearchkitRequest'
import BaseQuery from '../query/BaseQuery'
import { BaseFacet } from '../facets/BaseFacet'

export interface SortingOption {
  id: string
  label: string
  field: any
  defaultOption: boolean
}

export interface SearchkitConfig {
  host: string
  index: string
  sortOptions?: SortingOption[]
  hits: {
    fields: string[]
  }
  query?: BaseQuery
  facets?: Array<BaseFacet>
}

export interface ResultsResolverParameters {
  filters: Array<MixedFilter>
  query: string
}

export default (config: SearchkitConfig) => async (parent, parameters, ctx) => {
  try {
    const skConfig = {
      sortOptions: [],
      ...config
    }
    const queryManager = new QueryManager(parameters.filters, parameters.query)
    const skRequest = new SearchkitRequest(queryManager, skConfig)

    ctx.searchkit = {
      skRequest: skRequest,
      queryManager: queryManager,
      config: skConfig
    }

    return {}
  } catch (e) {
    console.log(e)
  }
}
