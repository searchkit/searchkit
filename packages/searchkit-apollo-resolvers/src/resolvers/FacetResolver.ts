import SearchkitRequest from '../core/SearchkitRequest'
import { getAggregationsFromFacets, getFacetsFromResponse } from '../core/FacetsFns'
import QueryManager from '../core/QueryManager'
import { SearchkitConfig } from './ResultsResolver'

export interface FacetResolverParameters {
  id: string
  query: string
  size: number
}

export default async (parent, parameters: FacetResolverParameters, ctx) => {
  const queryManager: QueryManager = ctx.searchkit.queryManager
  const config: SearchkitConfig = ctx.searchkit.config
  const skRequest: SearchkitRequest = ctx.searchkit.skRequest

  const facet = config.facets && config.facets.find((facet) => facet.getId() === parameters.id)
  if (!facet) return null

  try {
    const aggs = getAggregationsFromFacets(
      queryManager,
      {
        [facet.getId()]: {
          query: parameters.query,
          size: parameters.size
        }
      },
      [facet]
    )
    const response = await skRequest.search(aggs)
    const facets = getFacetsFromResponse([facet], response)

    return facets[0]
  } catch (e) {
    console.log(e)
  }
}
