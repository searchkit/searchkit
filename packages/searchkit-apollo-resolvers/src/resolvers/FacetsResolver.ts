import SearchkitRequest from '../core/SearchkitRequest'
import { getAggregationsFromFacets, getFacetsFromResponse } from '../core/FacetsFns'
import QueryManager from '../core/QueryManager'
import { SearchkitConfig } from './ResultsResolver'

export default async (parent, {}, ctx) => {
  const queryManager: QueryManager = ctx.searchkit.queryManager
  const config: SearchkitConfig = ctx.searchkit.config
  const skRequest: SearchkitRequest = ctx.searchkit.skRequest
  if (!config.facets) return null

  try {
    const aggs = getAggregationsFromFacets(queryManager, {}, config.facets)
    const results = await skRequest.search(aggs)
    const facets = getFacetsFromResponse(config.facets, results)

    return facets
  } catch (e) {
    console.log(e)
  }
}
