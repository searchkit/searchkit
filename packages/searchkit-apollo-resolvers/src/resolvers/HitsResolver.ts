import SearchkitRequest from '../core/SearchkitRequest'
import { SearchkitConfig, SortingOption } from './ResultsResolver'

export interface HitsParameters {
  page: {
    from: number
    size: number
  }
  sortBy?: string
}

function getSortOption(id, sortOptions: SortingOption[]) {
  const selectedSortOption = sortOptions.find((sortOption) => sortOption.id === id)
  if (!selectedSortOption) {
    throw new Error(`Sort Option: sorting option ${id} not found`)
  }
  return selectedSortOption
}

export default async (parent, parameters: HitsParameters, ctx) => {
  const config: SearchkitConfig = ctx.searchkit.config
  const skRequest: SearchkitRequest = ctx.searchkit.skRequest
  try {
    const from = parameters.page?.from || 0
    const size = parameters.page?.size || 10
    const defaultSortOption = config.sortOptions.find((so) => so.defaultOption)

    const chosenSortOption = parameters.sortBy
      ? getSortOption(parameters.sortBy, config.sortOptions)
      : defaultSortOption

    const { hits } = await skRequest.search({
      from: from,
      size: size,
      sort: chosenSortOption ? chosenSortOption.field : [{ _score: 'desc' }]
    })

    return {
      items: hits.hits.map((hit) => ({
        id: hit._id,
        fields: hit._source
      })),
      page: {
        total: hits.total.value,
        totalPages: Math.ceil(hits.total.value / size),
        pageNumber: from / size,
        from: from,
        size: size
      },
      sortedBy: chosenSortOption?.id
    }
  } catch (e) {
    console.log(e)
  }
}
