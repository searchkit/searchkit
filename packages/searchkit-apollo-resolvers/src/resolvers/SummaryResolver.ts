import SearchkitRequest from '../core/SearchkitRequest'
import QueryManager from '../core/QueryManager'
import { SearchkitConfig } from './ResultsResolver'

export default async (parent, {}, ctx) => {
  const queryManager: QueryManager = ctx.searchkit.queryManager
  const config: SearchkitConfig = ctx.searchkit.config
  const skRequest: SearchkitRequest = ctx.searchkit.skRequest

  try {
    const results = await skRequest.search({})
    return {
      total: results.hits.total.value ? results.hits.total.value : results.hits.total,
      query: queryManager.getQuery(),
      sortOptions: config.sortOptions.map((sortOption) => ({
        id: sortOption.id,
        label: sortOption.label
      })),
      appliedFilters: queryManager.getFilters().map((filterSet) => {
        const facetConfig = config.facets.find((facet) => facet.getId() === filterSet.id)
        return {
          label: facetConfig.getLabel(),
          id: facetConfig.getId(),
          value: filterSet.value
        }
      }, [])
    }
  } catch (e) {
    console.log(e)
  }
}
