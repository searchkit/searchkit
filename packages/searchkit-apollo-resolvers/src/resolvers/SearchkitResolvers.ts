import ResultsResolver, { SearchkitConfig } from './ResultsResolver'
import HitsResolver from './HitsResolver'
import FacetsResolver from './FacetsResolver'
import SummaryResolver from './SummaryResolver'
import FacetResolver from './FacetResolver'

export default (config: SearchkitConfig) => ({
  ResultSet: {
    hits: HitsResolver,
    facets: FacetsResolver,
    facet: FacetResolver,
    summary: SummaryResolver
  },
  Query: {
    results: ResultsResolver(config)
  },
  FacetSet: {
    __resolveType: (e) => e.type
  },
  SelectedFilter: {
    __resolveType: (e) => e.type
  }
})
