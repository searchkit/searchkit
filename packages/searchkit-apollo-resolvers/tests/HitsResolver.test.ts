import { SearchkitConfig } from '../src/resolvers/ResultsResolver'
import { MultiMatchQuery } from '../src'
import { setupTestServer, callQuery } from './support/helper'

describe('Hits Resolver', () => {
  describe('should return as expected', () => {
    const runQuery = async (query = '', page = { size: 10, from: 0 }, sorting?: string) => {
      const gql = `
        {
          results(query: "${query}") {
            hits(page: {size: ${page.size}, from: ${page.from}} ${
        sorting ? `, sortBy: "${sorting}"` : ''
      }) {
              sortedBy
              items {
                id
                fields {
                  writers
                  actors
                }
              }
            }
          }
        }
      `
      const response = await callQuery({ gql })
      return response
    }

    it('should return correct Results', async () => {
      const config: SearchkitConfig = {
        host: 'http://localhost:9200',
        index: 'movies',
        hits: {
          fields: ['actors', 'writers']
        },
        query: new MultiMatchQuery({ fields: ['actors', 'writers', 'title^4', 'plot'] })
      }

      setupTestServer(config)

      const response = await runQuery()
      expect(response.body.data).toMatchSnapshot()
      expect(response.status).toEqual(200)
    })

    it('should return correct Results on page 2', async () => {
      const config: SearchkitConfig = {
        host: 'http://localhost:9200',
        index: 'movies',
        hits: {
          fields: ['actors', 'writers']
        },
        query: new MultiMatchQuery({ fields: ['actors', 'writers', 'title^4', 'plot'] })
      }

      setupTestServer(config)

      const response = await runQuery('', { size: 10, from: 10 })
      expect(response.body.data).toMatchSnapshot()
      expect(response.status).toEqual(200)
    })

    it('should provide sorting options', async () => {
      const config: SearchkitConfig = {
        host: 'http://localhost:9200',
        index: 'movies',
        hits: {
          fields: ['actors', 'writers']
        },
        sortOptions: [
          { id: 'relevance', label: 'Relevance', field: '_score' },
          { id: 'released', label: 'Recent Releases', field: { released: 'desc' } }
        ],
        query: new MultiMatchQuery({ fields: ['actors', 'writers', 'title^4', 'plot'] })
      }

      setupTestServer(config)

      let response = await runQuery('', { size: 10, from: 10 }, 'relevance')
      expect(response.body.data).toMatchSnapshot()

      response = await runQuery('', { size: 10, from: 10 }, 'released')
      expect(response.body.data).toMatchSnapshot()

      expect(response.status).toEqual(200)
    })
  })
})
