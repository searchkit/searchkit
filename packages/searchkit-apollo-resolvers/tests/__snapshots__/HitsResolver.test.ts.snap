// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits Resolver should return as expected should provide sorting options 1`] = `
Object {
  "results": Object {
    "hits": Object {
      "items": Array [
        Object {
          "fields": Object {
            "actors": Array [
              "Al Pacino",
              "Robert Duvall",
              "Diane Keaton",
              "Robert De Niro",
            ],
            "writers": Array [
              "Francis Ford Coppola",
              "Mario Puzo",
            ],
          },
          "id": "tt0071562",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Liam Neeson",
              "Ben Kingsley",
              "Ralph Fiennes",
              "Caroline Goodall",
            ],
            "writers": Array [
              "Thomas Keneally",
              "Steven Zaillian",
            ],
          },
          "id": "tt0108052",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Tom Hanks",
              "Rebecca Williams",
              "Sally Field",
              "Michael Conner Humphreys",
            ],
            "writers": Array [
              "Winston Groom",
              "Eric Roth",
            ],
          },
          "id": "tt0109830",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Bryan Cranston",
              "Anna Gunn",
              "Aaron Paul",
              "Dean Norris",
            ],
            "writers": Array [
              "Vince Gilligan",
            ],
          },
          "id": "tt0903747",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Bruce Allpress",
              "Sean Astin",
              "John Bach",
              "Sala Baker",
            ],
            "writers": Array [
              "J.R.R. Tolkien",
              "Fran Walsh",
              "Philippa Boyens",
              "Stephen Sinclair",
              "Peter Jackson",
            ],
          },
          "id": "tt0167261",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Mark Hamill",
              "Harrison Ford",
              "Carrie Fisher",
              "Peter Cushing",
            ],
            "writers": Array [
              "George Lucas",
            ],
          },
          "id": "tt0076759",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Morgan Freeman",
              "Andrew Kevin Walker",
              "Daniel Zacapa",
              "Brad Pitt",
            ],
            "writers": Array [
              "Andrew Kevin Walker",
            ],
          },
          "id": "tt0114369",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Mark Hamill",
              "Harrison Ford",
              "Carrie Fisher",
              "Billy Dee Williams",
            ],
            "writers": Array [
              "Leigh Brackett",
              "Lawrence Kasdan",
              "George Lucas",
            ],
          },
          "id": "tt0080684",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Christian Bale",
              "Gary Oldman",
              "Tom Hardy",
              "Joseph Gordon-Levitt",
            ],
            "writers": Array [
              "Jonathan Nolan",
              "Christopher Nolan",
              "David S. Goyer",
              "Bob Kane",
            ],
          },
          "id": "tt1345836",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Robert De Niro",
              "Ray Liotta",
              "Joe Pesci",
              "Lorraine Bracco",
            ],
            "writers": Array [
              "Nicholas Pileggi",
              "Martin Scorsese",
            ],
          },
          "id": "tt0099685",
        },
      ],
      "sortedBy": "relevance",
    },
  },
}
`;

exports[`Hits Resolver should return as expected should provide sorting options 2`] = `
Object {
  "results": Object {
    "hits": Object {
      "items": Array [
        Object {
          "fields": Object {
            "actors": Array [
              "Channing Tatum",
              "Jamie Foxx",
              "Maggie Gyllenhaal",
              "Jason Clarke",
            ],
            "writers": Array [
              "James Vanderbilt",
            ],
          },
          "id": "tt2334879",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Brad Pitt",
              "Mireille Enos",
              "Daniella Kertesz",
              "James Badge Dale",
            ],
            "writers": Array [
              "Matthew Michael Carnahan",
              "Drew Goddard",
              "Damon Lindelof",
              "J. Michael Straczynski",
              "Max Brooks",
            ],
          },
          "id": "tt0816711",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Greta Gerwig",
              "Mickey Sumner",
              "Michael Esper",
              "Adam Driver",
            ],
            "writers": Array [
              "Noah Baumbach",
              "Greta Gerwig",
            ],
          },
          "id": "tt2347569",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Henry Cavill",
              "Amy Adams",
              "Michael Shannon",
              "Diane Lane",
            ],
            "writers": Array [
              "David S. Goyer",
              "Christopher Nolan",
              "Jerry Siegel",
              "Joe Shuster",
            ],
          },
          "id": "tt0770828",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Amy Acker",
              "Alexis Denisof",
              "Nathan Fillion",
              "Clark Gregg",
            ],
            "writers": Array [
              "Joss Whedon",
              "William Shakespeare",
            ],
          },
          "id": "tt2094064",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "James Franco",
              "Jonah Hill",
              "Seth Rogen",
              "Jay Baruchel",
            ],
            "writers": Array [
              "Seth Rogen",
              "Evan Goldberg",
              "Jason Stone",
            ],
          },
          "id": "tt1245492",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Matt Damon",
              "Scott Bakula",
              "Eric Zuckerman",
              "Eddie Jemison",
            ],
            "writers": Array [
              "Richard LaGravenese",
              "Scott Thorson",
              "Alex Thorleifson",
            ],
          },
          "id": "tt1291580",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Vince Vaughn",
              "Owen Wilson",
              "Rose Byrne",
              "Aasif Mandvi",
            ],
            "writers": Array [
              "Vince Vaughn",
              "Jared Stern",
            ],
          },
          "id": "tt2234155",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Emilia Clarke",
              "Kit Harington",
              "Richard Madden",
              "Iain Glen",
            ],
            "writers": Array [
              "George R.R. Martin",
              "David Benioff",
              "D.B. Weiss",
            ],
          },
          "id": "tt2178784",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Sherri Saum",
              "Teri Polo",
              "Jake T. Austin",
              "David Lambert",
            ],
            "writers": Array [
              "Brad Bredeweg",
              "Peter Paige",
            ],
          },
          "id": "tt2262532",
        },
      ],
      "sortedBy": "released",
    },
  },
}
`;

exports[`Hits Resolver should return as expected should return correct Results 1`] = `
Object {
  "results": Object {
    "hits": Object {
      "items": Array [
        Object {
          "fields": Object {
            "actors": Array [
              "Tim Robbins",
              "Morgan Freeman",
              "Bob Gunton",
              "William Sadler",
            ],
            "writers": Array [
              "Stephen King",
              "Frank Darabont",
            ],
          },
          "id": "tt0111161",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Marlon Brando",
              "Al Pacino",
              "James Caan",
              "Richard S. Castellano",
            ],
            "writers": Array [
              "Mario Puzo",
              "Francis Ford Coppola",
            ],
          },
          "id": "tt0068646",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Christian Bale",
              "Heath Ledger",
              "Aaron Eckhart",
              "Michael Caine",
            ],
            "writers": Array [
              "Jonathan Nolan",
              "Christopher Nolan",
              "David S. Goyer",
              "Bob Kane",
            ],
          },
          "id": "tt0468569",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Peter Dinklage",
              "Lena Headey",
              "Maisie Williams",
              "Emilia Clarke",
            ],
            "writers": Array [
              "David Benioff",
              "D.B. Weiss",
            ],
          },
          "id": "tt0944947",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Tim Roth",
              "Amanda Plummer",
              "Laura Lovelace",
              "John Travolta",
            ],
            "writers": Array [
              "Quentin Tarantino",
              "Roger Avary",
            ],
          },
          "id": "tt0110912",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Edward Norton",
              "Brad Pitt",
              "Helena Bonham Carter",
              "Meat Loaf",
            ],
            "writers": Array [
              "Chuck Palahniuk",
              "Jim Uhls",
            ],
          },
          "id": "tt0137523",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Noel Appleby",
              "Alexandra Astin",
              "Sean Astin",
              "David Aston",
            ],
            "writers": Array [
              "J.R.R. Tolkien",
              "Fran Walsh",
              "Philippa Boyens",
              "Peter Jackson",
            ],
          },
          "id": "tt0167260",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Leonardo DiCaprio",
              "Joseph Gordon-Levitt",
              "Ellen Page",
              "Tom Hardy",
            ],
            "writers": Array [
              "Christopher Nolan",
            ],
          },
          "id": "tt1375666",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Alan Howard",
              "Elijah Wood",
              "Noel Appleby",
              "Sean Astin",
            ],
            "writers": Array [
              "J.R.R. Tolkien",
              "Fran Walsh",
              "Philippa Boyens",
              "Peter Jackson",
            ],
          },
          "id": "tt0120737",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Keanu Reeves",
              "Laurence Fishburne",
              "Carrie-Anne Moss",
              "Hugo Weaving",
            ],
            "writers": Array [
              "Andy Wachowski",
              "Lana Wachowski",
            ],
          },
          "id": "tt0133093",
        },
      ],
      "sortedBy": null,
    },
  },
}
`;

exports[`Hits Resolver should return as expected should return correct Results on page 2 1`] = `
Object {
  "results": Object {
    "hits": Object {
      "items": Array [
        Object {
          "fields": Object {
            "actors": Array [
              "Al Pacino",
              "Robert Duvall",
              "Diane Keaton",
              "Robert De Niro",
            ],
            "writers": Array [
              "Francis Ford Coppola",
              "Mario Puzo",
            ],
          },
          "id": "tt0071562",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Liam Neeson",
              "Ben Kingsley",
              "Ralph Fiennes",
              "Caroline Goodall",
            ],
            "writers": Array [
              "Thomas Keneally",
              "Steven Zaillian",
            ],
          },
          "id": "tt0108052",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Tom Hanks",
              "Rebecca Williams",
              "Sally Field",
              "Michael Conner Humphreys",
            ],
            "writers": Array [
              "Winston Groom",
              "Eric Roth",
            ],
          },
          "id": "tt0109830",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Bryan Cranston",
              "Anna Gunn",
              "Aaron Paul",
              "Dean Norris",
            ],
            "writers": Array [
              "Vince Gilligan",
            ],
          },
          "id": "tt0903747",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Bruce Allpress",
              "Sean Astin",
              "John Bach",
              "Sala Baker",
            ],
            "writers": Array [
              "J.R.R. Tolkien",
              "Fran Walsh",
              "Philippa Boyens",
              "Stephen Sinclair",
              "Peter Jackson",
            ],
          },
          "id": "tt0167261",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Mark Hamill",
              "Harrison Ford",
              "Carrie Fisher",
              "Peter Cushing",
            ],
            "writers": Array [
              "George Lucas",
            ],
          },
          "id": "tt0076759",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Morgan Freeman",
              "Andrew Kevin Walker",
              "Daniel Zacapa",
              "Brad Pitt",
            ],
            "writers": Array [
              "Andrew Kevin Walker",
            ],
          },
          "id": "tt0114369",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Mark Hamill",
              "Harrison Ford",
              "Carrie Fisher",
              "Billy Dee Williams",
            ],
            "writers": Array [
              "Leigh Brackett",
              "Lawrence Kasdan",
              "George Lucas",
            ],
          },
          "id": "tt0080684",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Christian Bale",
              "Gary Oldman",
              "Tom Hardy",
              "Joseph Gordon-Levitt",
            ],
            "writers": Array [
              "Jonathan Nolan",
              "Christopher Nolan",
              "David S. Goyer",
              "Bob Kane",
            ],
          },
          "id": "tt1345836",
        },
        Object {
          "fields": Object {
            "actors": Array [
              "Robert De Niro",
              "Ray Liotta",
              "Joe Pesci",
              "Lorraine Bracco",
            ],
            "writers": Array [
              "Nicholas Pileggi",
              "Martin Scorsese",
            ],
          },
          "id": "tt0099685",
        },
      ],
      "sortedBy": null,
    },
  },
}
`;
